gtsam/discrete/DecisionTreeFactor.cpp

```diff
@@ -286,5 +286,45 @@ namespace gtsam {
         AlgebraicDecisionTree<Key>(keys, table),
         cardinalities_(keys.cardinalities()) {}
 
+  /* ************************************************************************ */
+  DecisionTreeFactor::shared_ptr DecisionTreeFactor::prune(
+      size_t maxNrLeaves) const {
+    const size_t N = maxNrLeaves;
+
+    // Let's assume that the structure of the last discrete density will be the
```

Comment:
seems to refer to application context. What is "last density" ?

Reply:
@ProfFan I think you added this, I'll update it.

Reply:
This is also resolved in the other PR (borglab 1151).